### PR TITLE
test margin on maxspan compared to interval and window

### LINKED ISSUE #1796: [Bug] EQL maxspan test doesn't account for from-to
**Describe the bug**
The test logic created in #1343 will fail on rule configured like the below:
```
from = "now-25h"
interval = "15m"

sequence by user.name with maxspan=24h
```

image

**To Reproduce**
1. create a new EQL sequence rule
2. Apply the fields/config as described
3. run `python -m detection_rules test
4. See the error

**Expected behavior**
The test takes all three of the parameters into account to determine if there is a (significant) chance for missing events in the monitor/query configuration.

**Additional context**
Discussion with @rw-access on Slack:
https://elasticstack.slack.com/archives/C016E72DWDS/p1645790032366879

### PR DESCRIPTION
**Pull request by SHolzhauer**
## Issues
Closes #1796

## Summary
Adding logic to test for margin on long sequences

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

**Comment by SHolzhauer** [comment]
Let me know if you have any feedback, hope this is passable for code style :sweat_smile:

**Comment by Mikaayenson** [comment]
@SHolzhauer Thanks for submitting the PR! There appears to be formatting changes for visual purposes rather that logical changes, which is one of the issues failing in the PR check. Do you mind reviewing [python build errors](https://github.com/elastic/detection-rules/runs/5345777627?check_suite_focus=true) and resolve the issues?

**Comment by brokensound77** [comment]
Thanks for the contribution @SHolzhauer. Can you revert all of the unrelated format changes for this

**Review by Mikaayenson — APPROVED** [review]
Made small visual edits to address the flake8 issues. Other than that, LGTM.

**Review by Mikaayenson — APPROVED** [review]

**Comment by SHolzhauer** [comment]
@brokensound77 @Mikaayenson Anything else needed on this PR?

**Review by Mikaayenson — APPROVED** [review]
Looking at this a little closer, I think we can reduce the check from `margin <  2 * interval` to `margin < interval` for recommended windows to be increased to the shortest time.

**Review by terrancedejesus — APPROVED** [review]
Logic and math makes sense based on our conversation. Tested the handful of rules, altering their time values to match against recommendations from the tests and they appear fine.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -225,6 +228,21 @@ Did not set the following values because they are un-required when set to the de
 ```
 
 
+### Understanding interval, lookback, and maxspan for EQL sequence rules
+
+Refer to the [docs](https://www.elastic.co/guide/en/security/current/rules-api-create.html) to see the definitions of these terms.
```
Comment (Mikaayenson) [inline CONTRIBUTING.md:233]: This link does not include definitions for maxspan or lookback.

**Comment by brokensound77** [comment]
This is still under review

**Pull request closed without merging**